server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleIndexer.java

```diff
@@ -136,4 +225,40 @@ private static IndexRequest newIndexRequest(RuleExtensionDoc ruleExtension) {
       .source(ruleExtension.getFields())
       .parent(ruleExtension.getParent());
   }
+
+  private static IndexRequest newRuleDocIndexRequest(RuleForIndexingDto ruleForIndexingDto) {
+    RuleDoc ruleDoc = RuleDoc.of(ruleForIndexingDto);
+
+    return new IndexRequest(INDEX_TYPE_RULE.getIndex(), INDEX_TYPE_RULE.getType(), ruleDoc.key().toString())
+      .source(ruleDoc.getFields());
+  }
+
+  private static IndexRequest newRuleExtensionDocIndexRequest(RuleForIndexingDto ruleForIndexingDto) {
+    RuleExtensionDoc ruleExtensionDoc = RuleExtensionDoc.of(ruleForIndexingDto);
+
+    return new IndexRequest(INDEX_TYPE_RULE_EXTENSION.getIndex(), INDEX_TYPE_RULE_EXTENSION.getType(), ruleExtensionDoc.getId())
+      .source(ruleExtensionDoc.getFields())
+      .parent(ruleExtensionDoc.getParent());
+  }
+
+  private static IndexRequest newRuleExtensionDocIndexRequest(RuleExtensionForIndexingDto ruleExtensionForIndexingDto) {
+    RuleExtensionDoc ruleExtensionDoc = RuleExtensionDoc.of(ruleExtensionForIndexingDto);
+
+    return new IndexRequest(INDEX_TYPE_RULE_EXTENSION.getIndex(), INDEX_TYPE_RULE_EXTENSION.getType(), ruleExtensionDoc.getId())
+      .source(ruleExtensionDoc.getFields())
+      .parent(ruleExtensionDoc.getParent());
+  }
+
+  private BulkIndexer newBulkIndexerForRules(Size bulkSize) {
+    return new BulkIndexer(esClient, INDEX_TYPE_RULE.getIndex(), bulkSize);
+  }
+
+  private static ListMultimap<EsQueueDto.Type, EsQueueDto> groupItemsByType(Collection<EsQueueDto> items) {
+    return items.stream().collect(MoreCollectors.index(EsQueueDto::getDocType));
+  }
+
+  private static RuleExtensionId explodeRuleExtensionDocId(EsQueueDto esQueueDto) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "groupRulesExtensionByOrganization" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)